polygon/p2p/fetcher_base.go

```diff
@@ -25,12 +25,12 @@ type FetcherConfig struct {
 
 type Fetcher interface {
 	// FetchHeaders fetches [start,end) headers from a peer. Blocks until data is received.
-	FetchHeaders(ctx context.Context, start uint64, end uint64, peerId *PeerId) ([]*types.Header, error)
+	FetchHeaders(ctx context.Context, start uint64, end uint64, peerId *PeerId) ([]*types.Header, int, error)
```

Comment:
would suggest to have FetchHeaders return `FetcherResponse[[]*types.Header, error`
same for FetchBodies, FetchBlocks